.github/actions/dockerhub_login/action.yml

```diff
@@ -16,20 +16,40 @@ inputs:
   key_password:
```

Comment:
this is a preparation to move the docker image build in to ci-test.yml -- so auto builds can create a new base image if needed....  need to figure out a tagging/naming strategy that doesn't cause side effects for master/release branches though.